invenio_app_rdm/records_ui/views/decorators.py

```diff
@@ -34,25 +36,53 @@ def pass_record(f):
     """Decorate a view to pass a record using the record service."""
     @wraps(f)
     def view(**kwargs):
-        pid_value = kwargs.get('pid_value')
-        record = service().read(
-            id_=pid_value, identity=g.identity, links_config=links_config()
-        )
-        kwargs['record'] = record
-        return f(**kwargs)
+        try:
+            pid_value = kwargs.get('pid_value')
+            record = service().read(
+                id_=pid_value, identity=g.identity, links_config=links_config()
+            )
+            kwargs['record'] = record
+            return f(**kwargs)
+        except PermissionDeniedError:
+            if not current_user.is_authenticated:
+                return current_app.login_manager.unauthorized()
+            abort(403)
     return view
 
 
 def pass_draft(f):
     """Decorator to retrieve the draft using the record service."""
     @wraps(f)
     def view(**kwargs):
-        pid_value = kwargs.get('pid_value')
-        draft = service().read_draft(
-            id_=pid_value,
-            identity=g.identity,
-            links_config=draft_links_config()
-        )
-        kwargs['draft'] = draft
-        return f(**kwargs)
+        try:
+            pid_value = kwargs.get('pid_value')
+            draft = service().read_draft(
+                id_=pid_value, identity=g.identity, links_config=links_config()
+            )
+            kwargs['draft'] = draft
+            return f(**kwargs)
+        except PermissionDeniedError:
```

Comment:
You can use ``blueprint.error_handler()``  ([see example](https://github.com/inveniosoftware/invenio-records-ui/blob/master/invenio_records_ui/views.py#L66-L72)). This way we'll handle it on all views in the same manner. We'll anyway also need it for PIDDoesNotExists error etc.